CDCgov/NEDSS-Modernization
# CNFT2-1153 options api

## Description

Adds an Options API to retrieve lists of items in a standard format.  The `concepts` endpoints will return concepts formatted so that they can be directly used by `SelectInput` and `MultiSelectInput`. 

There are two endpoints available;

GET `/nbs/api/options/concepts/{value-set}`

Returns ALL Concepts within a `value-set`

GET `/nbs/api/options/concepts/{value-set}/search?criteria={criteria}&limit={limit}`

Returns any Concepts up to the `limit` within a `value-set` that have a name starting with the `criteria`.  The `limit` is optional and defaults to `15` to match the [USWDS guidance](https://designsystem.digital.gov/components/select/) for when to use a Select component or a Combo Box.

The API was added in the `modernization-api` since it is a general use API.  

- Updates the `modernization-api` Postman collection to include the new API
- Adds the `useConceptOptions` hook to interact with the API that returns all concepts
- Adds the `useConceptOptionAutocomplete` hook to interact with the autocomplete API.

## Tickets

* [CNFT2-1153](https://cdc-nbs.atlassian.net/browse/CNFT2-1153)

## Steps to verify

1. Import the `modernization-api` Postman collection
2. Start the `modernization-api` using `./gradlew modernization-api:bootRun`
3. Ensure that the `NDESS-Modernization-API > Options > Concepts > All Concepts by Value Set` requests returns the expected results.
4.  Ensure that the `NDESS-Modernization-API > Options > Concepts > Specific Concepts by Value Set` requests returns the expected results.


[CNFT2-1153]: https://cdc-nbs.atlassian.net/browse/CNFT2-1153?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
